### PR TITLE
Task/36 - create navigation component

### PR DESCRIPTION
**Pull request by magda2307**

### CODE REVIEW

File: app/src/main/java/com/safetyheads/akademiaandroida/dashboard_placeholder.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+private const val ARG_PARAM2 = "param2"
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [dashboard_placeholder.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class dashboard_placeholder : Fragment() {
+    // TODO: Rename and change types of parameters
+    private var param1: String? = null
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/dashboard_placeholder.kt:21]: you can totally remove the param1 and param2 thing

File: app/src/main/java/com/safetyheads/akademiaandroida/dashboard_placeholder.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+         * Use this factory method to create a new instance of
+         * this fragment using the provided parameters.
+         *
+         * @param param1 Parameter 1.
+         * @param param2 Parameter 2.
+         * @return A new instance of fragment dashboard_placeholder.
+         */
+        // TODO: Rename and change types and number of parameters
+        @JvmStatic
+        fun newInstance(param1: String, param2: String) =
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/dashboard_placeholder.kt:51]: this approach is very good, however please remove params

File: app/src/main/java/com/safetyheads/akademiaandroida/dashboard_placeholder.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
+package com.safetyheads.akademiaandroida
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+
+// TODO: Rename parameter arguments, choose names that match
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/dashboard_placeholder.kt:9]: remove unnecessary comments after you remove the params

File: app/src/main/java/com/safetyheads/akademiaandroida/dashboard_placeholder.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
+private const val ARG_PARAM2 = "param2"
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [dashboard_placeholder.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class dashboard_placeholder : Fragment() {
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/dashboard_placeholder.kt:19]: class name should be DashboardPlaceholder

File: app/src/main/java/com/safetyheads/akademiaandroida/login_placeholder.kt [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import android.widget.Button
+import androidx.fragment.app.Fragment
+import androidx.navigation.Navigation
+
+// TODO: Rename parameter arguments, choose names that match
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
+private const val ARG_PARAM2 = "param2"
+
+class login_placeholder : Fragment() {
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/login_placeholder.kt:16]: LoginPlaceholder

File: app/src/main/java/com/safetyheads/akademiaandroida/login_placeholder.kt [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import androidx.fragment.app.Fragment
+import androidx.navigation.Navigation
+
+// TODO: Rename parameter arguments, choose names that match
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/login_placeholder.kt:13]: the same comments as in DashboardPlaceholder

File: app/src/main/java/com/safetyheads/akademiaandroida/mainscreen_placeholder.kt [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import androidx.navigation.Navigation
+
+// TODO: Rename parameter arguments, choose names that match
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/mainscreen_placeholder.kt:11]: the same comments as in DashboardPlaceholder

File: app/src/main/java/com/safetyheads/akademiaandroida/mainscreen_placeholder.kt [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
+private const val ARG_PARAM2 = "param2"
+
+
+class mainscreen_placeholder : Fragment() {
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/mainscreen_placeholder.kt:17]: MainScreenPlaceholder

File: app/src/main/java/com/safetyheads/akademiaandroida/notlogged_placeholder.kt [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+
+// TODO: Rename parameter arguments, choose names that match
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/notlogged_placeholder.kt:11]: The same comments as in DashboardPlaceholder

File: app/src/main/java/com/safetyheads/akademiaandroida/notlogged_placeholder.kt [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+private const val ARG_PARAM2 = "param2"
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [notlogged_placeholder.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class notlogged_placeholder : Fragment() {
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/notlogged_placeholder.kt:19]: NotLoggedPlaceholder

File: app/src/main/java/com/safetyheads/akademiaandroida/splash_placeholder.kt [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
+private const val ARG_PARAM2 = "param2"
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [splash_placeholder.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class splash_placeholder : Fragment() {
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/splash_placeholder.kt:19]: SplashPlaceholder

File: app/src/main/java/com/safetyheads/akademiaandroida/splash_placeholder.kt [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+
+// TODO: Rename parameter arguments, choose names that match
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/splash_placeholder.kt:11]: the same comments as in DashboardPlaceholder

File: app/src/main/java/com/safetyheads/akademiaandroida/login_placeholder.kt [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        val view = inflater.inflate(R.layout.fragment_login_placeholder, container, false)
+        val loginButton = view.findViewById<Button>(R.id.button_login)
+        loginButton.setOnClickListener {
+            Navigation.findNavController(view).navigate(R.id.dashboard_placeholder)
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/login_placeholder.kt:37]: R.id.action_login_placeholder_to_dashboard_placeholder should be used instead 🤔 
this applies to all ...navigate() calls you have

Comment (przemek19980102) [reply app/src/main/java/com/safetyheads/akademiaandroida/login_placeholder.kt:37]: use actions that are in XML instead of calling fragments directly

**Review by przemek19980102 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/safetyheads/akademiaandroida/RootActivity.kt [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+        Navigation.setViewNavController(findViewById(R.id.nav_host_fragment), navController)
+        val splashTime = 3000L // time in milliseconds
+        //splash->main after specified time
+        Handler(Looper.getMainLooper()).postDelayed({
+            navController.navigate(R.id.mainscreen_placeholder)
+        }, splashTime)
+
+        val onBackPressedCallback = object : OnBackPressedCallback(true) {
+            override fun handleOnBackPressed() {
+                if (navController.currentDestination?.id == R.id.mainscreen_placeholder || navController.currentDestination?.id == R.id.dashboard_placeholder) {
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/RootActivity.kt:32]: zerknij proszę na https://stackoverflow.com/a/59426267 czy da się w XMLu zastąpić tego OnBackPressedCallback?

**Review by przemek19980102 — CHANGES_REQUESTED** [review]

**Review by przemek19980102 — DISMISSED** [review]

File: app/src/main/AndroidManifest.xml [hunk 15]
```diff
@@ -17,12 +17,15 @@
         <activity
             android:name=".TabLayoutActivity"
             android:exported="true">
+        </activity>
+        <activity android:name=".RootActivity"
```
Comment (msarelo) [inline app/src/main/AndroidManifest.xml:21]: Zamiast zmieniać startową aktywność, dodaj przycisk który odpali ten fragment z MainActivity

**Review by msarelo — CHANGES_REQUESTED** [review]
Rozwiąż konflikty

**Pull request merged**